core/trino-parser/src/test/java/io/trino/sql/parser/ParserAssert.java

```diff
@@ -73,15 +73,22 @@ public ParserAssert ignoringLocation()
         return ignoringFieldsMatchingRegexes("(.*\\.)?location");
     }
 
-    private static <T extends Node> AssertProvider<ParserAssert> createAssertion(Function<String, T> parser, String sql)
+    private static <T extends Node> AssertProvider<ParserAssert> createAssertion(Function<String, T> parseOperation, String sql)
     {
-        return () -> new ParserAssert(parser.apply(sql), newRecursiveComparisonConfig())
+        final T actualStatement = parseOperation.apply(sql);
+
+        return () -> new ParserAssert(actualStatement, newRecursiveComparisonConfig())
                 .withRepresentation(NODE_REPRESENTATION)
-                .satisfies(parsed -> new ParserAssert(parser.apply(formatSql((Node) parsed)), newRecursiveComparisonConfig())
-                        .describedAs("Validate SQL->AST->SQL roundtrip")
-                        .withRepresentation(NODE_REPRESENTATION)
-                        .ignoringLocation()
-                        .isEqualTo(parser.apply(sql)));
+                .satisfies(parsedStatement -> {
+                    final String parsedFormattedSql = formatSql((Node) parsedStatement);
+                    final T parsedFormattedStatement = parseOperation.apply(parsedFormattedSql);
+
+                    new ParserAssert(parsedFormattedStatement, newRecursiveComparisonConfig())
+                            .describedAs("Validate SQL->AST->SQL roundtrip")
+                            .withRepresentation(NODE_REPRESENTATION)
+                            .ignoringLocation()
+                            .isEqualTo(actualStatement);
+                });
```

Comment:
Sure thing. I still think it is worth keeping the change at line 90 however, since we'd otherwise be needlessly parsing the query string again (old line 78 and 84)?

Reply:
That's ok. Please put it in a separate commit, since it's unrelated to the main purpose of this commit.